lib/event.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+import socket
+
+from utilities.toUTC import toUTC
+
+
+class Event(dict):
+
+    # We set default vaules so that we can later
+    # create an alert around these, and know when events
+    # have to use defaults
+    DEFAULT_STRING = 'None'
+
+    def add_required_fields(self):
+        if 'receivedtimestamp' not in self:
```

Comment:
These values are the required fields that we have listed in the docs. Do we want to make 'source', 'timestamp', and details, all required fields?

Reply:
timestamp - is it always a copy of the UTC timestamp, or it can be anything that the source sets? Could it be a local (for the source) time in a local timezone?
source - if we can come up with some smart heuristics
details - source creates it, otherwise it's not present. I'm not sure we want to have it set to null, for example, because it's kind of like a pointer to a JSON dict